applications/tari_app_grpc/src/conversions/peer.rs

```diff
@@ -0,0 +1,106 @@
+//
+// Redistribution and use in source and binary forms, with or without modification, are permitted provided that the
+// following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following
+// disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use crate::tari_rpc as grpc;
+use std::collections::HashMap;
+use tari_comms::{
+    net_address::MutliaddrWithStats,
+    peer_manager::{Peer, PeerFeatures, PeerFlags},
+};
+use tari_crypto::tari_utilities::ByteArray;
+
+impl From<Peer> for grpc::Peer {
+    fn from(peer: Peer) -> Self {
+        let public_key = Vec::from(peer.public_key.as_bytes());
+        let node_id = peer.node_id.short_str();
+        let mut addresses: Vec<grpc::Address> = Vec::new();
+        let last_connection = match peer.addresses.last_seen() {
+            Some(v) => v.to_string(),
+            None => "".to_string(),
+        };
+        for address in peer.addresses.addresses {
+            addresses.push(address.clone().into())
+        }
+        let flags = if peer.flags == PeerFlags::empty() {
+            "".to_string()
+        } else {
+            format!("{:?}", peer.flags)
+        };
+        let banned_until = match peer.banned_until {
+            Some(v) => v.to_string(),
+            None => "".to_string(),
+        };
+        let banned_reason = peer.banned_reason.to_string();
+        let offline_at = match peer.offline_at {
+            Some(v) => v.to_string(),
+            None => "".to_string(),
+        };
+        let features = match peer.features {
+            PeerFeatures::COMMUNICATION_NODE => "BASE_NODE".to_string(),
+            PeerFeatures::COMMUNICATION_CLIENT => "WALLET".to_string(),
+            f => format!("{:?}", f),
+        };
```

Comment:
The receiver of the peer data should make the base node/wallet determination - in future we could have validator node in the features (though likely it's easier to just have a PeerRole enum). 
 
```suggestion
        let features = match peer.features.bits();
          
```